Moving S390X to SOC7

aplanas opened the pull request.

aplanas committed:
nova: add some documented defaults in template for s390x

aplanas committed:
nova: drop zvm_xcat_network parameter

nicolasbock commented on chef/data_bags/crowbar/template-nova.json:
```diff
@@ -86,14 +86,13 @@
-        "zvm_scsi_pool": "",
-        "zvm_user_profile": "",
+        "zvm_scsi_pool": "xcatzfcp",
+        "zvm_user_profile": "OSDFLT",
```
Where do those come from?

dirkmueller commented:
@aplanas rebase?

aplanas committed:
cinder: enable LVM volumes with DASD devices

aplanas commented:
Requires https://github.com/crowbar/crowbar-core/pull/1351

aplanas commented:
![](https://github.trello.services/images/mini-trello-icon.png) [P2: Adapt Cinder barclamp to detect zVM name schema](https://trello.com/c/ZSak1h6X/136-p2-adapt-cinder-barclamp-to-detect-zvm-name-schema)

aplanas committed:
nova: update `compute_driver` parameter for s390x

aplanas committed:
nova: improve disk type default for s390x

Most of the time the type of disk will be ECKD.  This also fix
a bug in the selection list.

aplanas committed:
nova: select the s390x node by default for compute role

If a node is based on s390x architecture, the nova-compute-zvm
role is assigned automatically.

Add a NOTE to remind the drop of this feature for SOC8 if the
default hypervisor is zKVM.

aplanas committed:
nova: add basic validation for s390x

aplanas committed:
nova: do not show infrequent attributes

aplanas committed:
neutron: add validation for VLAN with zVM

If a node in the cloud is a zVM node, we will require (for now)
to include VLAN as a valid ML2 agent too.

aplanas committed:
neutron: add additional_zvm_networks attribute for s390x

Some zVM deployments can have multiple additional networks, that are
automatically created by the neutron-zvm-agent.  Each network will have
a VSWITCH in the zVM side (also created by the same agent).  Those
networks can be flat (vlan unaware) or vlan (vlan aware).

The new attribute will store a list of networks with this schema:

  name[:[vlan_min]:[vlan_max][:rdev_list]

This will populate the attributes flat_networs and network_vlan_ranges
from ml2_conf.ini file, and will define sections in neutron_zvm_plugin
for each new network.

aplanas committed:
neutron: hide deprecated parameters for s390x

xcat_mgt_ip and xcat_mgt_mask are deprecated in Neutron.  We hide
them from the UI, and remove the defaults in the databag.

aplanas committed:
neutron: hide rdev parameter for s390x

rsalevsky commented:
Do we still need this?

aplanas commented:
yep
